### PR TITLE
simplify Run in controllermanager.go

### PR DESCRIPTION
**Pull request by fisherxu**
**What this PR does / why we need it**:
remove part of code in controllermanager.go to createclient,createRecorder,startHTTP function

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -116,47 +116,15 @@ func Run(s *options.CMServer) error {
 	} else {
 		glog.Errorf("unable to register configz: %s", err)
 	}
-	kubeconfig, err := clientcmd.BuildConfigFromFlags(s.Master, s.Kubeconfig)
+
+	kubeClient, leaderElectionClient, kubeconfig, err := createClient(s)
```
Comment (wojtek-t) [inline cmd/kube-controller-manager/app/controllermanager.go]: s/createClient/createClients/

**Review by wojtek-t — COMMENTED** [review]
one minor nit - other than that LGTM.

File: cmd/kube-controller-manager/app/controllermanager.go (hunk 1 above)
Comment (fisherxu) [reply cmd/kube-controller-manager/app/controllermanager.go]: @wojtek-t  Have addressed, thanks  :)

**Comment by fisherxu** [comment]
Thanks :) @xiangpengzhao  @wojtek-t

**Pull request merged**